docs/book/src/tasks/experimental-features/cluster-classes.md

```diff
@@ -0,0 +1,497 @@
+clusterclass-quickstart-XXXX                                   clusterclass-quickstart   2m48s   docker:////clusterclass-quickstart-XXXX                                  Running   v1.21.2
+
+```
+
+To update the Cluster the only change needed is to the 'version' field under spec.topology in the Cluster object. Change '1.21.2' to '1.22' as below. 
+```yaml
+apiVersion: cluster.x-k8s.io/v1alpha4
+kind: Cluster
+metadata:
+  name: "clusterclass-quickstart"
+  namespace: default
+  labels:
+    cni: kindnet
+spec:
+  clusterNetwork:
+    services:
+      cidrBlocks: ["10.128.0.0/12"]
+    pods:
+      cidrBlocks: ["192.168.0.0/16"]
+    serviceDomain: "cluster.local"
+  topology:
+    class: cluster-class
+    version: v1.22.0
+      controlPlane:
+      metadata:
+        labels:
+          topologyLabel: topologyLabelValue
+        annotations:
+          topologyAnnotation: topologyAnnotation
+      replicas: 1 
+    workers:
+      machineDeployments:
+        - class: linux-worker
+          name: linux-workers
+          metadata:
+            labels:
+              topologyLabel: topologyLabelValue
+            annotations:
+              topologyAnnotation: topologyAnnotation
+          replicas: 1 
+```
+
+Save the above change to your local file and apply the update to the Cluster with:
+```bash
+kubectl apply -f clusterclass.yaml
+```
+
+The upgrade will take a minute to roll out as it will take place machine by machine with older versions of the machines only being removed after health newer versions come online.
+
+To watch the update progress run:
+```bash
+watch kubectl get  kubeadmcontrolplane,machines
+```
+After a few minutes the upgrade will be complete and the output will be similar to:
+```bash
+NAME                                                                              AGE   INITIALIZED   API SERVER AVAILABLE   VERSION   REPLICAS   READY   UPDATED   UNAVAILABLE
+kubeadmcontrolplane.controlplane.cluster.x-k8s.io/clusterclass-quickstart-XXXX    25m   true          true                   v1.22.0   1          1       1
+
+NAME                                                                                    CLUSTER                   AGE   PROVIDERID                                                                PHASE     VERSION
+machine.cluster.x-k8s.io/clusterclass-quickstart-linux-workers-XXXX                     clusterclass-quickstart   10m   docker:////clusterclass-quickstart-linux-workers-XXXX                     Running   v1.22.0
+machine.cluster.x-k8s.io/clusterclass-quickstart-XXXX                                   clusterclass-quickstart   13m   docker:////clusterclass-quickstart-skdmm-XXXX                             Running   v1.22.0
+```
+
+### Clean Up
+
+Delete workload cluster.
+```bash
+kubectl delete cluster capi-quickstart
```

Comment:
Note: this will only work if you rename the cluster (as suggested) to `capi-quickstart`.